dsc/workflows/base/__init__.py

```diff
@@ -134,6 +175,73 @@ def reconcile_bitstreams_and_metadata(self) -> bool:
         )
         """
 
+    @final
+    def write_reconcile_results_to_dynamodb(
+        self, item_identifier: str, status: str, run_date: datetime
+    ) -> None:
+        """Write reconcile results for each item submission to DynamoDB.
+
+        An item submission is first recorded in the DynamoDB table during
+        the 'reconcile' step. Once a record is successfully created or
+        retrieved from the table, this method will perform a series of
+        checks to determine how/what to update for the item submission
+        record. If the current status indicates that the item submission
+        is at a step that comes after 'reconcile', the record is not
+        updated in DynamoDB.
+        """
+        item_submission_log_details = (
+            f"with primary keys item_identifier={item_identifier} (hash key) and "
+            f"batch_id={self.batch_id} (range key)"
+        )
+
+        try:
+            item_submission_record = ItemSubmissionDB.create(
+                item_identifier=item_identifier,
+                batch_id=self.batch_id,
+                workflow_name=self.workflow_name,
+            )
+        except ItemSubmissionExistsError as exception:
+            logger.info(exception)
+            try:
+                item_submission_record = ItemSubmissionDB.get(
+                    hash_key=self.batch_id, range_key=item_identifier
+                )
+            except DoesNotExist as exception:
+                logger.error(  # noqa: TRY400
+                    f"Failed to retrieve record {item_submission_log_details}"
+                )
+                return
+        except PutError as exception:
+            logger.error(  # noqa: TRY400
+                f"Unable to create record {item_submission_log_details}: "
+                f"{exception.cause_response_message}"
+            )
+            return
```

Comment:
Could this be refactored to the `ItemSubmissionDB` model, something like `get_or_create()`?

That would keep this method more focused on setting the right status and saving it.